docs/Manifests.md

```diff
@@ -1,6 +1,190 @@
 # Manifests
 
-## AppManifest
+## AppManifest Specification
+
+The AppManifest YAML file defines your application's components, routes, resources, and lifecycle events.
+
+### Components top-level element
+
+An AppManifest must declare a `components` root element as a list of objects containing configuration for each component in your application. At least one component object must be defined.
+
+```yaml
+components:
+  - name: webserver
+    ...
+  - name: api
+    ...
+```
+
+<!-- <details> -->
+
+#### name
+
+Defines the name of the component. Each component `name` must be unique. **Required**.
+
+#### type
+
+Specifies the type of the component; must be one of `service`, `task`, or `static`. **Required**.
+
+#### image
+
+Sets the base image from which the component is built. **Required**.
+
+#### port
+
+The container port to expose.
+
+#### root
+
+Establishes the docker build context, setting the host working directory for `copy` and `run` build step commands.
```

Comment:
Open question for @arbitrarytech on if/how we want to make direct connections to Docker throughout our documentation. Obviously Docker is inherently a core aspect of Noop under-the-hood, but I think it might be good to position our documentation in a way that can still be understood by someone who might not have experience with Docker.